src/mpid/ch4/netmod/ofi/ofi_types.h

```diff
@@ -132,30 +132,6 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_idata_get_error_bits(uint64_t idata)
 #define MPIDI_OFI_DATATYPE(dt)   ((dt)->dev.netmod.ofi)
 #define MPIDI_OFI_COMM(comm)     ((comm)->dev.ch4.netmod.ofi)
 
-/* Convert the address vector entry to an endpoint index.
- * This conversion depends on the data structure which could change based on
- * whether we're using scalable endpoints or not. */
-MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_av_to_ep(MPIDI_OFI_addr_t * av)
-{
-#if MPIDI_OFI_ENABLE_ENDPOINTS_BITS
-    return (av)->ep_idx;
```

Comment:
This deletion is the major conflict with our downstream code. We have lots of done work relying `ep_idx` to explore multi endpoints usage.

Reply:
I am trying to understand why we need it. In upstream code, this `ep_idx` is always `0`. Practically, when we need a sub-endpoint for a particular vni, the `ep_idx` is just the `vni` (or deterministic from `vni`), right? If you have more scalable end points beyond vni, then you'll need some code for the derivation anyway, since a single `ep_idx` here won't help you distinguish different sub-endpoint anyway, right?

Reply:
In our downstream code, RMA uses an additional scalable endpoint. Its transmit contexts (for RMA) may map to the same vni as pt2pt. That means `vni` is not `deterministic` enough to determine which `ep` in the backend.

Reply:
Interesting. So you don't use dedicated `rma_sep` for RMA? In upstream main code, we use `rma_sep` and store the context id per-window (inside `win`) so there isn't need to store `ep_idx` elsewhere. In https://github.com/pmodels/mpich/pull/4770, I am expanding this  `rma_sep` to be per-vni. 

I am still a bit confused even in the case we use global `sep`. When you map `RMA` to the same vni as pt2pt, are you using the same transmission context or separate one for RMA? Either way, the `ep_idx` in the `av` only can point to the  same `context`, so you will need extra code to determine the actual `ep_idx` anyway, right?

Reply:
We use dedicated `rma_sep`, but contexts for RMA are bound to the same group of completion queues as `pt2pt`. And our code refers idx for completion queues same as `vni` idx.

In this case, let's say we have both `pt2pt_sep` and `rma_sep`, so:
ctx0 in `pt2pt_sep` is bound to cq 0, which is vni0
ctx0 in `rma_sep` is bound to cq 0, which is vni0.

I think your design is another alternative for supporting multi-contexts in a particular `rma_sep`, maybe in future our downstream code will switch to the same as main. (we also don't want keep too much difference in our own branch).

Last questions above, from the description I just said, `RMA` uses a separate transmission (when scalable endpoint is supported). 

Also, one point in our code is that, the same `av` is bound to `pt2pt_sep` and `rma_sep`, in that case, looking for an `addr` in `av` doesn't require `ep_idx`. I think you are right to delete `ep_idx` for `av_lookup`.

```
MPIDI_OFI_CALL_RETURN(fi_scalable_ep_bind
                          (MPIDI_OFI_global.rma_sep, &(MPIDI_OFI_global.ctx[0].av->fid), 0), ret);
```

Let me forward this information to our team, to see if I mis-undestood anything.

Reply:
Is there any resolution to this discussion? I see this PR is waiting on a review from me, but I think the downstream conflicts need to be understood first.

Reply:
@sagarth ?

Reply:
I'm waiting on @wesbland to see if this change conflicts his downstream multi-nic mapping.

Reply:
Oh, sorry. I guess I should tag @shawnccx ?

Reply:
Yeah, this isn't as big a deal for the multi-nic stuff since that's already part of the address information. The deletion of `MPIDI_OFI_av_insert` has more impact there, but since it isn't used much, I can work around that. This is up to @shawnccx.

Reply:
Then I think we can move forward.